pkg/agent/clean/node_windows.go

```diff
@@ -0,0 +1,293 @@
+package clean
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/sirupsen/logrus"
+)
+
+const ContainerName = "cattle-node-cleanup"
+
+func Job() error {
+	logrus.Infof("Starting clean container job: %s", ContainerName)
+
+	c, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	ctx := context.Background()
+
+	containerList, err := c.ContainerList(ctx, types.ContainerListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, c := range containerList {
+		for _, n := range c.Names {
+			if n == "/"+ContainerName {
+				logrus.Infof("container named %s already exists, exiting.", ContainerName)
+				return nil
+			}
+		}
+	}
+
+	binds := []string{
+		"\\\\.\\pipe\\docker_engine:\\\\.\\pipe\\docker_engine",
+		"c:\\:c:\\host",
+		"\\\\.\\pipe\\rancher_wins:\\\\.\\pipe\\rancher_wins",
+	}
+
+	container, err := c.ContainerCreate(ctx, &container.Config{
+		Image: getAgentImage(),
+		Env: []string{
+			"AGENT_IMAGE=" + getAgentImage(),
+			"PREFIX_PATH=" + os.Getenv("PREFIX_PATH"),
+			"WINDOWS_PREFIX_PATH=" + os.Getenv("WINDOWS_PREFIX_PATH"),
+		},
+		Cmd: []string{"--", "agent", "clean"},
+	}, &container.HostConfig{
+		Binds: binds,
+	}, &network.NetworkingConfig{}, ContainerName)
+
+	if err != nil {
+		return err
+	}
+
+	return c.ContainerStart(ctx, container.ID, types.ContainerStartOptions{})
+}
+
+func Node() error {
+	logrus.Info("Cleaning up node...")
+
+	c, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	ctx := context.Background()
+	if err := waitForK8sPods(ctx, c); err != nil {
+		return fmt.Errorf("error waiting for k8s pods to be removed: %s", err)
+	}
+
+	return spawnCleanup()
+}
+
+func spawnCleanup() error {
+	if err := writeScript(); err != nil {
+		return err
+	}
+	winsArgs := createWinsArgs("Spawn")
+	output, err := exec.Command("wins.exe", winsArgs...).Output()
+	if err != nil {
+		logrus.Infof(string(output))
+		return err
+	}
+	return nil
+}
+
+func writeScript() error {
+	// add a null file to the container for wins to find and make a hash
+	psPath := getPowershellPath()
+	if !fileExists(psPath) {
+		psHostPath := strings.Replace(psPath, "c:\\", "c:\\host\\", 1)
```

Comment:
maybe this should be in a helper function `getHostPathFromContainer` (returns the string with host prepended) and `getContainerPath` (returns the exact string back) to make it obvious to read?

Would also be nice if we can add `getHostPathFromHost` which handles the prefixPath logic there itself.